modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/processing/Processing.scala

```diff
@@ -380,4 +382,19 @@ object Processing {
       a.countBytes
   }
 
+  private def possiblyExitOnMissingIgluSchema[F[_]: Sync](
+    env: Environment[F],
+    v2NonAtomicFields: NonAtomicFields.Result,
+    legacyFields: LegacyColumns.Result
+  ): F[Unit] =
+    if (env.exitOnMissingIgluSchema && (v2NonAtomicFields.igluFailures.nonEmpty || legacyFields.igluFailures.nonEmpty)) {
+      val base =
+        "Exiting because failed to resolve Iglu schemas.  Either check the configuration of the Iglu repos, or set the `skipSchemas` config option, or set `exitOnMissingIgluSchema` to false.\n"
+      val failures = v2NonAtomicFields.igluFailures.map(_.failure) ::: legacyFields.igluFailures.map(_.failure)
+      val msg      = failures.map(_.asJson.noSpaces).mkString(base, "\n", "")
+      Logger[F].error(base) *> env.appHealth.beUnhealthyForRuntimeService(RuntimeService.Iglu) *> Sync[F].raiseError(
```

Comment:
That's very much a detail but I think that setting the app unhealthy should be the very first thing to do, before logging